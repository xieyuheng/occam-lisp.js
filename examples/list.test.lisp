(import nil li length append "list.lisp")
(import true "bool.lisp")
(import zero add1 "nat-church.lisp")

(assert-equal (length nil) zero)
(assert-equal (length (li true nil)) (add1 zero))
(assert-equal (length (li true (li true nil))) (add1 (add1 zero)))

(assert-equal (append nil nil) nil)
(assert-equal (append nil (li true nil)) (li true nil))
(assert-equal (append (li true nil) nil) (li true nil))
(assert-equal (append (li true nil) (li true nil)) (li true (li true nil)))
(assert-equal (append (li true (li true nil))
                      (li true (li true nil)))
              (li true (li true (li true (li true nil)))))
