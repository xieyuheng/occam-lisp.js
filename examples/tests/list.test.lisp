['a]
['a 'b 'c]
[1 2 3]
['a 'b 'c 1 2 3 "a b c"]

[[1 2 3]
 [4 5 6]
 [7 8 9]]

(assert
  (equal?
   [[1 2 3]
    [4 5 6]
    [7 8 9]]
   [[1 2 3]
    [4 5 6]
    [7 8 9]]))

(assert
  (not
   (equal?
    [[1 2 3]
     [4 5 6]
     [7 8 9]]
    [[1 2 3]
     [4 5 6]
     [7 8 10]])))

(assert
  (equal?
   (list-length [1 2 3])
   3))

(assert
  (equal?
   (list-append [1 2 3] [4 5 6])
   [1 2 3 4 5 6]))

(assert
  (equal?
   (list-of [1 2 3 :x 1 :y 2 :z 3])
   [1 2 3]))
