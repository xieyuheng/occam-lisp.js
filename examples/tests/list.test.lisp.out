['a]
['a 'b 'c]
[1 2 3]
['a 'b 'c 1 2 3 "a b c"]
[[1 2 3] [4 5 6] [7 8 9]]
[assert] fail
  exp: (equal? (list-length [1 2 3]) 2)
[source] examples/tests/list.test.lisp:29:1
 27|     [7 8 10]])))
 28|
 29|(assert
    ~~~~~~~~
 30|  (equal?
    ~~~~~~~~~~
 31|   (list-length [1 2 3])
    ~~~~~~~~~~~~~~~~~~~~~~~~~
 32|   2))
    ~~~~~~ 
 33|
 34|(assert

